src/proposals/20241017/SparkEthereum_20241017.t.sol

```diff
@@ -0,0 +1,240 @@
+
+        assertEq(allConfigsAfter.length, 12);
+
+        ReserveConfig memory susds = ReserveConfig({
+            symbol:                  'sUSDS',
+            underlying:               SUSDS,
+            aToken:                   address(0),  // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            variableDebtToken:        address(0),  // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            stableDebtToken:          address(0),  // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+            decimals:                 18,
+            ltv:                      79_00,
+            liquidationThreshold:     80_00,
+            liquidationBonus:         105_00,
+            liquidationProtocolFee:   10_00,
+            reserveFactor:            10_00,
+            usageAsCollateralEnabled: true,
+            borrowingEnabled:         false,
+            interestRateStrategy:     _findReserveConfigBySymbol(allConfigsAfter, 'sUSDS').interestRateStrategy,
+            stableBorrowRateEnabled:  false,
+            isPaused:                 false,
+            isActive:                 true,
+            isFrozen:                 false,
+            isSiloed:                 false,
+            isBorrowableInIsolation:  false,
+            isFlashloanable:          false,
+            supplyCap:                50_000_000,
+            borrowCap:                0,
+            debtCeiling:              0,
+            eModeCategory:            0
+        });
+
+        _validateReserveConfig(susds, allConfigsAfter);
+
+        _validateInterestRateStrategy(
+            susds.interestRateStrategy,
+            susds.interestRateStrategy,
+            InterestStrategyValues({
+                addressesProvider:             address(poolAddressesProvider),
+                optimalUsageRatio:             0.8e27,
+                optimalStableToTotalDebtRatio: 0,
+                baseStableBorrowRate:          0.02e27,
+                stableRateSlope1:              0,
+                stableRateSlope2:              0,
+                baseVariableBorrowRate:        0,
+                variableRateSlope1:            0.02e27,
+                variableRateSlope2:            3e27
+            })
+        );
+        
+        _assertSupplyCapConfig({
+            asset:            SUSDS,
+            max:              500_000_000,
+            gap:              50_000_000,
+            increaseCooldown: 12 hours
+        });
+
+        _assertBorrowCapConfig({
+            asset:            SUSDS,
+            max:              0,
+            gap:              0,
+            increaseCooldown: 0
+        });
+
+        // The sUSDS price feed does not have a decimals() function so we validate manually
+        IAaveOracle oracle = IAaveOracle(poolAddressesProvider.getPriceOracle());
+
+        require(
```

Comment:
what's the reason for using a rquire instead of `assertEq`?__

Reply:
There is a function to test this called `_validateAssetSourceOnOracle`. As the oracle doesn't have the decimals function it does not work for this. 

I opted for copy-pasting the block of code exactly as it was in the function to minimize any possible errors and to have it as similar as possible

Reply:
oh right, I did not know about that function, instead was afraid of there being a difference in behaviour between `assertEq` and `require` this test relied on.